### ISSUE TITLE
It seems to be unstable when using PyTorch with nvidia-dali for loading dataset saved on HDD

### ISSUE THREAD

**Issue by ipScore** [NVIDIA/DALI#1524]
Environment:
Ubuntu 16.04, PyTorch 1.2, nvidia-dali 0.15.0, Single RTX2080ti GPU.
Dataset:
**ILSVRC2012** saved on a **HDD**.
Code:
[ShuffleNetV2](https://github.com/megvii-model/ShuffleNet-Series/tree/master/ShuffleNetV2)
Params: 
model-size=0.5x, batch_size=256, num_workers=8

Pipe code:
```
class HybridTrainPipeline(Pipeline):
    def __init__(self, image_dir, batch_size, num_threads, device_id, crop_size):
        super(HybridTrainPipeline, self).__init__(batch_size, num_threads, device_id, seed=12+device_id)
        self.input = ops.FileReader(file_root=image_dir, random_shuffle=True)
        self.decode = ops.ImageDecoder(
            device="mixed",
            output_type=types.RGB,
            device_memory_padding=211025920,
            host_memory_padding=140544512,
        )
        self.res = ops.RandomResizedCrop(device="gpu", size=crop_size)
        self.coin = ops.CoinFlip(probability=0.5)
        self.cmnp = ops.CropMirrorNormalize(
            device="gpu",
            output_dtype=types.FLOAT,
            output_layout=types.NCHW,
            image_type=types.RGB,
        )

    def define_graph(self):
        # decode the images
        jpegs, labels = self.input(name="Reader")
        images = self.decode(jpegs)
        # Augmentation
        images = self.res(images)
        # Normalization & mirror: Mask for horizontal flip with p=0.5
        images = self.cmnp(images, mirror=self.coin())
        # images on GPU
        return images, labels
```

Dataloader code:
```
class DataIterator(object):

    def __init__(self, dataloader):
        self.dataloader = dataloader
        self.iterator = enumerate(self.dataloader)

    def next(self):
        try:
            _, data = next(self.iterator)
        except Exception:
            self.iterator = enumerate(self.dataloader)
            _, data = next(self.iterator)
        return data[0]["data"], data[0]["label"].squeeze().cuda().long()

TrainPipe = HybridTrainPipeline(
        image_dir=args.train_dir,
        batch_size=args.batch_size,
        num_threads=8,
        device_id=0,
        crop_size=224
    )
    TrainPipe.build()
    train_loader = DALIClassificationIterator([TrainPipe], size=1281167)
    train_dataprovider = DataIterator(train_loader)
```

At the beginning stage of training, everything seems  to be OK and the GPU-Util is about 80%.
![image](https://user-images.githubusercontent.com/30183023/69899023-b245d100-139b-11ea-93a6-f1c06de42be1.png)

However, the GPU-Util decreases to only 10%  after a while, and the training time for 20Iters is more than twice compared with the beginning stage.
![image](https://user-images.githubusercontent.com/30183023/69899059-17012b80-139c-11ea-97f9-346b7aaf101a.png)
![image](https://user-images.githubusercontent.com/30183023/69899066-42841600-139c-11ea-8b1c-7685f39f2381.png)

Is there some kind advice for me? Thanks.

**Comment by JanuszL** [comment]
Hi,
You can check https://github.com/NVIDIA/DALI/issues/839 or https://github.com/NVIDIA/DALI/issues/1415 describing similar symptoms.
It is due to read access speed from your drive. I guess you are not using SSD or RAID to store your training data. You should run top/htop/any other system monitor to see CPU and IO utilization.
This high speed at the begging is due to caching, as your OS is keeping data read from disk in RAM so when you read them again it already there. But it cannot store all of it (your RAM is not big enough or you are reading data for the first time). In that case, DALI cannot do much as it is about processing while reading data from disc is up to HW you have. You can read more about [disc cache here](https://blog.webnersolutions.com/understanding-disk-cache-in-linux).

**Comment by ipScore** [comment]
Hi, @JanuszL.  Thanks for your advice. Yes, my training data is stored using HDD instead of SSD. So how can I speed the process of reading data from a HDD disk？
![image](https://user-images.githubusercontent.com/30183023/69926032-b84bc700-14ed-11ea-9bf7-8e8dad0a3041.png)

**Comment by JanuszL** [comment]
@ipScore - I don't think you can do much except replacing the HDD to SSD or arranging some raid partition. The HDD throughput is the limiter in your case. DALI help in the case then the CPU is the bottleneck, however when you cannot read the data fast enough from the HDD there is no much that can be done.

**Comment by ipScore** [comment]
@JanuszL Thanks.
